io-engine/src/core/snapshot.rs

```diff
@@ -34,7 +35,11 @@ pub trait SnapshotOps {
         &self,
         snap_param: SnapshotParams,
     ) -> Result<(), Self::Error>;
+
+    // Get Snapshot Iterator
+    async fn get_snapshot(self) -> LvolSnapshotIter;
```

Comment:
```suggestion
    /// Get a Snapshot Iterator.
    async fn snapshot_iter(self) -> LvolSnapshotIter;
```

Reply:
Should this be `LvolSnapshotIter` here? Seems like it's tying the trait to the LvolSnapshot iterator but it should be a generic snapshop iterator?

Reply:
Yes it should be generic. modified. Thanks